app/src/main/java/com/conupods/OutdoorMaps/MarkerClickListener.java

```diff
@@ -0,0 +1,20 @@
+package com.conupods.OutdoorMaps;
+
+import com.google.android.gms.maps.GoogleMap;
+import com.google.android.gms.maps.model.Marker;
+
+public class MarkerClickListener implements GoogleMap.OnMarkerClickListener {
+    private GoogleMap mMap;
+    private BuildingInfoWindow mBuildingInfoWindow;
+
+    public MarkerClickListener(GoogleMap map, BuildingInfoWindow buildingInfoWindow) {
+        this.mMap = map;
+        this.mBuildingInfoWindow = buildingInfoWindow;
+    }
+
+    @Override
+    public boolean onMarkerClick(Marker marker) {
+        mMap.setInfoWindowAdapter(mBuildingInfoWindow);
+        return false;
+    }
+}
```

Comment:
I think this logic should be put inside the mapinitializer class, turning the onMarkerClick into a lambda function. You will have to modify the constructor of the MapInitializer to take in your buildingInformation object.